libraries/stdlib/samples/test/samples/collections/collections.kt

```diff
@@ -764,6 +764,16 @@ class Collections {
         }
     }
 
+    class Filtering {
+        @Sample
+        fun filter() {
+            val numbers: List<Int> = listOf(1, 2, 3, 4)
+            val evenNumbers = numbers.filter { number -> number.rem(2) == 0 }
```

Comment:
Two notes:
- better to use `%` infix operator instead of its named form `rem`
- short lambdas usually do not introduce name for its parameter and use default `it` instead.

It may be beneficial, however, to show both variants, so we can do the following: 
- use `it` in `filter` lambda
- write another expression with `filterNot` that uses explicit parameter name
- now this sample can be used for `filterNot` as well :)